cuqi/experimental/mcmc/_hmc.py

```diff
@@ -0,0 +1,401 @@
+import numpy as np
+import numpy as np
+from cuqi.experimental.mcmc import SamplerNew
+from cuqi.array import CUQIarray
+
+
+class NUTSNew(SamplerNew):
+    """No-U-Turn Sampler (Hoffman and Gelman, 2014).
+
+    Samples a distribution given its logpdf and gradient using a Hamiltonian
+    Monte Carlo (HMC) algorithm with automatic parameter tuning.
+
+    For more details see: See Hoffman, M. D., & Gelman, A. (2014). The no-U-turn
+    sampler: Adaptively setting path lengths in Hamiltonian Monte Carlo. Journal
+    of Machine Learning Research, 15, 1593-1623.
+
+    Parameters
+    ----------
+    target : `cuqi.distribution.Distribution`
+        The target distribution to sample. Must have logpdf and gradient method.
+        Custom logpdfs and gradients are supported by using a
+        :class:`cuqi.distribution.UserDefinedDistribution`.
+    
+    initial_point : ndarray
+        Initial parameters. *Optional*
```

Comment:
Since optional, specify what happens if the user does not specify the initial_point.